labs/a5.md

```diff
@@ -0,0 +1,336 @@
+
+  You need to add the `retry:n` option so that you retry a query as many times as possible but the total time to resolve a name, irrelevant of success or failure, takes less than **5** seconds.  What should the value of n be?
+6. If we preferred name resolution be done dynamically rather than using static entries in **/etc/hosts** what file do we need to edit and what is the line we should add?
+
+## :fire: This is fine :fire:
+This section will have you thinking like a sysadmin.  
+
+Each script **might** make changes to your network stack with the intent of damaging your machine's connectivity.  To confine the scope of the 'attacks', scripts will specifically try to alter your connectivity to `google.com` and `ocf.berkeley.edu`.  These scripts are **dangerous** so we are having you run them inside a disposable vm which you should install using the following steps.
+
+1. Install `vagrant` by either downloading it [here][vagrant-download] or using the appropriate package manager for your system.
+2. Install `virtualbox` by either downloading it [here][vbox-download] or using the appropriate package manager for your system.
+  
+   As a side note the installation and configuration scripts have been tested with  `vagrant 2.0.2` and `virtualbox 5.2.8`.  They have additionally been confirmed to work with `vagrant v1.9.x` and `virtualbox 5.1.34`.  Compatibility issues may arise with other versions, please try shifting to the above listed versions or come to office hours.
+   
+3. On your own local machine i.e. laptop or desktop, clone the repo found [here][decal-labs].
+4. Go into the vm directory `cd decal-labs/a5/vm`.
+5. Run `vagrant up`.  Vagrant will automatically bring up a virtual machine on your local machine according to the files in the `vm` directory.
+6. `ssh` into your machine by running `ssh vagrant@192.168.42.42`, if this prompts you for a password it should default to `vagrant`.
+7. Your vm should be configured and also have the `decal-labs` repo cloned to your home directory.  Scripts are numbered and located under `decal-labs/a5/scenarios`.
+8. **Launch scripts, one at a time, with sudo, i.e. `sudo python3 <script.py>`.**
+9. For each script. You will be doing a 2 step process.  **Only move onto another script, once you have finished resolving your current one.**
+    1. Analyze whether or not your connectivity has been damaged. If your stack has been damaged identify the issue or which part of your network is no longer functioning as intended.
+    2. If you concluded there was a problem, resolve the issue.  What commands did you use and how did you conclude things were fully functional again?
+
+   **For each step you must explain the tools you used and how you came to your  conclusions i.e.** 
+> I ran `example --pls --fix computer` and I noticed that line 3: `computer-is-broken` meant my machine was f*****.
+
+> This script damaged my ability to connect to google.com by poisoning my arp cache with bogus entries.
```

Comment:
This line isn't indented like the one above it when it gets rendered, try indenting both lines by 4 spaces and they should be aligned.